src/main/java/seedu/cafectrl/parser/Parser.java

```diff
@@ -130,49 +130,42 @@ private static Command prepareAdd(String arguments) {
         final Pattern addArgumentPatter = Pattern.compile(ADD_ARGUMENT_STRING);
         Matcher matcher = addArgumentPatter.matcher(arguments);
 
-        // Checks whether the overall pattern of add arguments is correct
-        if (!matcher.matches()) {
-            return new IncorrectCommand("Error: Incorrect format for the add command.\n"
-                    + AddDishCommand.MESSAGE_USAGE);
-        }
-
         try {
+            // Checks whether the overall pattern of add arguments is correct
+            if (!matcher.matches()) {
+                return new IncorrectCommand(Messages.INVALID_ADD_DISH_FORMAT_MESSAGE
+                        + AddDishCommand.MESSAGE_USAGE);
+            }
+
             // To retrieve specific arguments from arguments
             String dishName = matcher.group(DISH_NAME_MATCHER_GROUP_NUM);
             float price = Float.parseFloat(matcher.group(PRICE_MATCHER_GROUP_NUM));
             String ingredientsListString = matcher.group(INGREDIENT_LIST_MATCHER_GROUP_NUM);
 
-            IncorrectCommand incorrectCommand1 = checkNegativePrice(price);
-            if (incorrectCommand1 != null) {
-                return incorrectCommand1;
-            }
+            checkNegativePrice(price);
 
             // Capture the list of ingredients
             ArrayList<Ingredient> ingredients = new ArrayList<>();
 
-            IncorrectCommand incorrectCommand2 = ingredientParsing(ingredientsListString, ingredients);
-            if (incorrectCommand2 != null) {
-                return incorrectCommand2;
-            }
+            ingredientParsing(ingredientsListString, ingredients);
```

Comment:
Perhaps you could make it return the modified ingredients as it would be useful for JUnit test and debugging in the future

Reply:
ok that is a good idea! I have already made the changes